src/plans/D3MCompoundPlan.sol

```diff
@@ -128,14 +131,31 @@ contract D3MCompoundPlan is ID3MPlan {
         } else revert("D3MCompoundPlan/file-unrecognized-param");
         emit File(what, data);
     }
-    function file(bytes32 what, address data) external auth {
-        if (what == "tack") tack = InterestRateModelLike(data);
-        else if (what == "delegate") delegate = data;
-        else revert("D3MCompoundPlan/file-unrecognized-param");
-        emit File(what, data);
+
+    function addTack(address tack) external auth {
+        tacks[tack] = 1;
+        emit AddTack(tack);
+    }
+
+    function removeTack(address tack) external auth {
+        tacks[tack] = 0;
+        emit RemoveTack(tack);
+    }
+
+    function addDelegate(address delegate) external auth {
+        delegates[delegate] = 1;
+        emit AddDelegate(delegate);
+    }
+
+    function removeDelegate(address delegate) external auth {
+        delegates[delegate] = 0;
+        emit RemoveDelegate(delegate);
```

Comment:
I'm not sure about breaking away from our `file` pattern, what about something like:

```solidity
function file(bytes32 what, address data, uint256 set) external auth {
  require(set == 0 || set == 1, "D3MCompoundPlan/file-invalid-set");
  if (what == "tack") tacks[data] = set;
  else if (what == "delegate") delegates[data] = set;
  else revert(""D3MCompoundPlan/file-file-unrecognized-param");
  emit File(what, data, set);
}
```
or 
```solidity
    function file(bytes32 what, address data) external {
        uint256 set;
        if (what == "tack") {
            set = ~tacks[data] & 1;
            tacks[data] = set;
        } else if (what == "delegate") {
            set = ~delegates[data] & 1; 
            delegates[data] = set;
        } else revert("D3MCompoundPlan/file-unrecognized-param");
        emit File(what, data, set);
    }
```
or
```solidity
    function file(bytes32 what, address data) external {
        uint256 set;
        if (what == "tack") {
            set = (tacks[data] == 1) ? 0 : 1;
            tacks[data] = set;
        } else if (what == "delegate") {
            set = (delegates[data] == 1) ? 0 : 1;
            delegates[data] = set;
        } else revert("D3MCompoundPlan/file-unrecognized-param");
        emit File(what, data, set);
    }
```

Reply:
Yeah, I like your 1st option, though haven't seen we did that with a boolean-like value before. Let's see what others think.

Reply:
> though haven't seen we did that with a boolean-like value before.

@talbaneth i'm not sure what you mean by this?

and I guess in addition to the example Sam has here, the best argument against what I'm proposing here is our `rely`/`deny` structure...

If others are okay with this I'm fine leaving as-is

Reply:
I'd go for option 1 but changing `data` to `addr` and `set` to `data`. This is probably the most similar example we have: https://github.com/makerdao/dss-teleport/blob/master/src/TeleportRouter.sol#L85

Reply:
> I'd go for option 1 but changing `data` to `addr` and `set` to `data`. This is probably the most similar example we have: https://github.com/makerdao/dss-teleport/blob/master/src/TeleportRouter.sol#L85

yeah `addr` and `data` is better

Reply:
Done in [6754ed0](https://github.com/makerdao/dss-direct-deposit/pull/94/commits/6754ed0a1bf5a0fb481c75a5f4ce3dafd0c024cf)